src/frontend/js/components/AddressesManagement/useAddressesManagement.tsx

```diff
@@ -0,0 +1,135 @@
+import { messages } from 'components/AddressesManagement/index';
+import { AddressesMutateOptions, useAddresses } from 'hooks/useAddresses';
+import { useState } from 'react';
+import { useIntl } from 'react-intl';
+import { MutateOptions } from 'react-query';
+import * as Joanie from 'types/Joanie';
+import { Address } from 'types/Joanie';
+import { Maybe } from 'types/utils';
+import { HttpError } from 'utils/errors/HttpError';
+import { confirm } from 'utils/indirection/window';
+
+/**
+ * High-level helper to handle various `Address` CRUD with loading and error built-in.
+ */
+export function useAddressesManagement() {
+  const intl = useIntl();
+  const addresses = useAddresses();
+  const [error, setError] = useState<Maybe<string>>();
+  const [load, setLoad] = useState(false);
+
+  /**
+   * Update the provided address to promote it as main
+   *
+   * @param {Joanie.Address} address
+   */
+  const promoteAddress = (address: Joanie.Address) => {
+    if (!address.is_main) {
+      setLoad(true);
+      setError(undefined);
+      addresses.methods.update(
+        {
+          ...address,
+          is_main: true,
+        },
+        {
+          onSuccess: () => {
+            setLoad(false);
+          },
+          onError: () => {
+            setError(intl.formatMessage(messages.actionPromotion));
+            setLoad(false);
+          },
+        },
+      );
+    }
+  };
+
+  /**
+   * Ask the user to confirm his intention
+   * then make the request to delete the provided address
+   *
+   * @param {Joanie.Address} address
+   * @param {AddressesMutateOptions} options
+   */
+  const removeAddress = (
+    address: Joanie.Address,
+    options?: MutateOptions<void, HttpError, string>,
+  ) => {
+    // eslint-disable-next-line no-alert, no-restricted-globals
+    const sure = confirm(
+      intl.formatMessage(messages.deletionConfirmation, { title: address.title }),
+    );
+    setLoad(true);
+    setError(undefined);
+    if (!address.is_main && sure) {
+      setLoad(true);
+      addresses.methods.delete(address.id, {
+        onError: () => {
+          setError(intl.formatMessage(messages.actionDeletion));
+          setLoad(false);
+        },
+        onSuccess: () => {
+          setLoad(false);
+        },
+        ...options,
+      });
+    }
```

Comment:
```suggestion
    }
    if address.is_main shouldn't it return a specific error message?
```

Reply:
In theory we don't display the button for `is_main` address, but to prevent misusage of this component I can add a specific error :)